financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkaccountpicker/LinkAccountPickerViewModel.kt

```diff
@@ -168,55 +180,137 @@ internal class LinkAccountPickerViewModel @Inject constructor(
+            selectedAccountIds = accountIds,
+        )
+        updateLocalManifest { it.copy(activeInstitution = null) }
+        FinancialConnections.emitEvent(name = Name.ACCOUNTS_SELECTED)
+        navigationManager.tryNavigateTo(Destination.Success(referrer = PANE))
+    }
+
+    private fun handleUnsupportedNextPane(nextPane: Pane?) {
+        when (nextPane) {
+            PARTNER_AUTH -> {
+                eventTracker.logError(
+                    extraMessage = "Connecting a supportability account, but user shouldn't be able to.",
+                    error = UnclassifiedError("ConnectSupportabilityAccountError"),
+                    logger = logger,
+                    pane = PANE,
                 )
-                eventTracker.track(Click("click.repair_accounts", PANE))
             }
-
-            Pane.PARTNER_AUTH -> {
-                updateLocalManifest { it.copy(activeInstitution = account.institution) }
+            BANK_AUTH_REPAIR -> {
+                eventTracker.logError(
+                    extraMessage = "Connecting a repair account, but user shouldn't be able to.",
+                    error = UnclassifiedError("ConnectRepairAccountError"),
+                    logger = logger,
+                    pane = PANE,
+                )
+            }
+            null -> {
+                eventTracker.logError(
+                    extraMessage = "Selected connect account, but next pane is NULL.",
+                    error = UnclassifiedError("ConnectUnselectedAccountError"),
+                    logger = logger,
+                    pane = PANE,
+                )
             }
+            else -> {
+                // Nothing to log here
+            }
+        }
+
+        // Fall back to the institution picker
+        val destination = nextPane?.destination?.invoke(referrer = PANE) ?: InstitutionPicker(referrer = PANE)
+        navigationManager.tryNavigateTo(destination)
+    }
+
+    private fun logAccountClick(partnerAccount: PartnerAccount) {
+        val state = stateFlow.value
+        val payload = state.payload() ?: return
+        val isNewSelection = partnerAccount.id !in state.selectedAccountIds
 
-            else -> Unit
+        val event = FinancialConnectionsAnalyticsEvent.AccountSelected(
+            selected = isNewSelection,
+            isSingleAccount = payload.singleAccount,
+            accountId = partnerAccount.id,
+        )
+
+        eventTracker.track(event)
+    }
+
+    fun onAccountClick(partnerAccount: PartnerAccount) {
+        val updateRequired = createUpdateRequiredContent(
+            partnerAccount = partnerAccount,
+            partnerToCoreAuths = stateFlow.value.payload()?.partnerToCoreAuths,
+        )
+
+        if (updateRequired != null) {
+            logUpdateRequired(updateRequired)
+            presentUpdateRequiredSheet(updateRequired, referrer = PANE)
+            return
+        }
+
+        logAccountClick(partnerAccount)
```

Comment:
Should we log _every_ account click or just the ones that don’t result in the update-required pane being shown? (cc @kgaidis-stripe)

Reply:
I follow web, and I believe web also does every click

Particularly because the web design selects the account update required account (visually and in code)